x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicator_barchart_legend_action/indicator_barchart_legend_action.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, VFC } from 'react';
+import { EuiButtonIcon, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import { FilterIn } from '../../../query_bar/components/filter_in';
+import { FilterOut } from '../../../query_bar/components/filter_out';
+
+export interface IndicatorBarchartLegendActionProps {
+  /**
+   * Indicator
+   */
+  data: string;
+  /**
+   * Indicator field selected in the IndicatorFieldSelector component, passed to the {@link AddToTimeline} to populate the timeline.
+   */
+  field: string;
+}
+
+export const IndicatorBarchartLegendAction: VFC<IndicatorBarchartLegendActionProps> = ({
+  data,
+  field,
+}) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const popoverItems = [
+    <EuiContextMenuItem key="addToTimeline" icon="timeline" size="s">
+      Add to Timeline
+    </EuiContextMenuItem>,
+    <FilterIn data={data} field={field} contextMenu={true} />,
```

Comment:
this is what I like the least about this approach, I don't like passing a boolean to change the display of the component. There must be a more elegant way to do this...

Reply:
Why don't you like it? what if we want only "Filter In" somewhere? I think it is nice that it's not bundled together. If you want you can create a new component that is just a composition of these two, eg `FilterInOut` to avoid this minor duplication of props. If you are concerned with some repetition inside the component, maybe you can think of a component `<Filter type="in" />` or smth along these lines. Another option is encapsulate the repeated code in sub-components or hooks if needed

Reply:
it's the `contextMenu` prop that I didn't like, the solution you suggested in the other comment would fix that!